tests/aws/services/lambda_/test_lambda_integration_dynamodbstreams.py

```diff
@@ -486,14 +529,22 @@ def test_dynamodb_event_filter(
         snapshot,
         aws_client,
     ):
+        """Test event filtering for DynamoDB streams:
+        https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html#filtering-ddb
+
+        Slow against AWS taking ~2min per test case.
+
+        Test assumption: The first item MUST always match the filter and the second item CAN match the filter.
+        => This enables two-step testing (i.e., snapshots between inserts) but is unreliable and should be revised.
+        """
         function_name = f"lambda_func-{short_uid()}"
         table_name = f"test-table-{short_uid()}"
         max_retries = 50
 
         create_lambda_function(
             handler_file=TEST_LAMBDA_PYTHON_ECHO,
             func_name=function_name,
-            runtime=Runtime.python3_9,
+            runtime=Runtime.python3_12,
```

Comment:
/cc @Morijarti I'm bumping the runtime here together with significant test changes, so can can simply take over my changes upon rebasing

Reply:
Awesome, thank you